src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -654,6 +697,14 @@ void handlePuback(int pubackMessageId) {
   void handlePubcomp(int pubcompMessageId) {
 
     synchronized (this.connection) {
+      io.netty.handler.codec.mqtt.MqttMessage removedPacket = qos2outbound.remove(pubcompMessageId);
+
+      if (removedPacket == null) {
+        return;
```

Comment:
Ditto as above. Add a log warning here, because a PUBCOMP is arrived but we don't have the corresponding message-id in the qos2outbound queue ... it's a weird situation :-)